au/packs.hh

```diff
@@ -519,39 +518,35 @@ struct IsValidPack : stdx::conjunction<detail::IsPackOf<Pack, T>,
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // `AreElementsInOrder` implementation.
 
-namespace detail {
-
 template <template <class...> class Pack>
-constexpr bool are_consecutive_elements_in_order_for() {
-    return true;
-}
+struct AreElementsInOrder<Pack, Pack<>> : std::true_type {};
 
 template <template <class...> class Pack, typename T>
-constexpr bool are_consecutive_elements_in_order_for() {
-    return true;
-}
+struct AreElementsInOrder<Pack, Pack<T>> : std::true_type {};
 
 template <template <class...> class Pack, typename T1, typename T2, typename... Ts>
-constexpr auto are_consecutive_elements_in_order_for() {
-    return InOrderFor<Pack, T1, T2>::value &&
-           are_consecutive_elements_in_order_for<Pack, T2, Ts...>();
-}
+struct AreElementsInOrder<Pack, Pack<T1, T2, Ts...>>
+    : stdx::conjunction<InOrderFor<Pack, T1, T2>, AreElementsInOrder<Pack, Pack<T2, Ts...>>> {};
 
-template <std::size_t N>
-constexpr bool all_true(const std::array<bool, N> &values) {
-    for (auto i = 0u; i < values.size(); ++i) {
-        if (!values[i]) {
+namespace detail {
+
+constexpr bool all_true() { return true; }
+
+template <typename... Predicates>
+constexpr bool all_true(Predicates &&...values) {
```

Comment:
Comment: I guess we could use `std::all_of`, but that is only `constexpr` in C++20 and we want it in C++14.

Reply:
Good suggestion, but yeah, I think we'll need to roll our own for that reason.